maven-surefire-plugin/src/main/java/org/apache/maven/plugin/surefire/SurefirePlugin.java

```diff
@@ -880,4 +900,20 @@ protected final ForkNodeFactory getForkNode()
     {
         return forkNode;
     }
+
+    @Override
+    boolean verifyParameters() throws MojoFailureException, MojoExecutionException
+    {
+        if ( failOnFlakeCount < 0 )
```

Comment:
As in my previous comment, this must be in the abstract class where are all the warning check method calls made in the section:
```
        else
        {
            ensureEnableProcessChecker();
            convertDeprecatedForkMode();
            ensureWorkingDirectoryExists();
            ensureParallelRunningCompatibility();
            ensureThreadCountWithPerThread();
            warnIfUselessUseSystemClassLoaderParameter();
            warnIfDefunctGroupsCombinations();
            warnIfRerunClashes();
            warnIfWrongShutdownValue();
            warnIfNotApplicableSkipAfterFailureCount();
            warnIfIllegalTempDir();
            warnIfForkCountIsZero();
            printDefaultSeedIfNecessary();
        }
```

Reply:
See above, this wouldn't make sense because IntegrationTestMojo does not have `failOnFlakeCount`. That's in the VerifyMojo, which doesn't have the same base class.

Reply:
@oehme 
I am not talking about VerifyMojo, nothing but the SurefirePlugin class.